File: C#/Prefix Sum/724. Find Pivot Index.cs
/* 724. Find Pivot Index
Difficulty: Easy
Submission Link: https://leetcode.com/problems/find-pivot-index/submissions/1383570205/ */

public class Solution {
    public int PivotIndex(int[] nums) {
        int a=nums.Sum(),b=0;
        for(int i=0;i<nums.Length;i++){
            a-=nums[i];
            if(a==b){
            return i;
            }
            b+=nums[i];
        }
        return -1;
    }
}

File: JavaScript/Prefix Sum/724. Find Pivot Index.js
/* 724. Find Pivot Index
Difficulty: Easy
Submission Link: https://leetcode.com/problems/find-pivot-index/submissions/1383569519/
*/

var pivotIndex = function(nums) {
    let total = nums.reduce((a, b) => a + b, 0);
    let leftTotal = 0;

    for (let i = 0; i < nums.length; i++) {
        let rightTotal = total - leftTotal - nums[i];

        if (rightTotal === leftTotal) {
            return i;
        }

        leftTotal += nums[i];
    }

    return -1;    
};